Updates to OIE Embedded Authentication > SDK Use Cases (#2399)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md

### Step 2: Select the phone factor in the authenticator list

In this use case, only the **phone** factor is displayed in the list of authenticators, as shown in the following example page:

<div class="common-image-format">

![Displays a screenshot of a Select Authenticator page that includes a phone option and a Select button.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-screen-verify-nodejs.png)

</div>

> >**Note:** For the SDK sample application, your individual users must set up a phone number for phone verification to see this authenticator option in the page.

Reviewer: your individual users > each user

> When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone type, `{ authenticator: 'phone' }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, indicating that the SDK needs a phone verification method in order to send the verification code. The `nextStep` field includes the input for this method.

Reviewer: ```suggestion
When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone type, `{ authenticator: 'phone' }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, which indicates that the SDK needs a phone verification method in order to send the verification code. The `nextStep` field includes the input for this method.
```

```JavaScript
status, // IdxStatus.PENDING
  nextStep: {
    inputs, // [{ phoneFactor: 'SMS', ... }]
}
```

The next step is to redirect the user to a page to enter in the phone verification field.

### Step 3: Build the phone verification method entry page

> Build the phone verification method entry page that accepts either SMS or voice verification that that is used for authentication.

Reviewer: ```suggestion
Build the phone verification method entry page that accepts either SMS or voice verification that is used for authentication.
```

<div class="common-image-format">

![Displays a screenshot of a Verify using phone authenticator page that includes a SMS option and a Next button.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-screen-verify-phone-method-nodejs.png)

</div>

> When the user enters a phone verification method, either SMS or voice verification, and clicks **Next**, a call to `idx.authenticate` is made that passes in the following values:

Reviewer: ```suggestion
When the user enters a phone verification method (either SMS or voice verification) and clicks **Next**, a call to `idx.authenticate` is made that passes in the following values:
```

Build a page that accepts the code sent to the user's phone number through SMS. Depending on your implementation, this page can be the same page that verifies the email code or a different page.

<div class="common-image-format">

![Displays a screenshot of a Challenge Authenticator page that includes a field for the code and a Verifiy button.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-in-pwd-phone-verify-phone-code-nodejs.png)

</div>

The next step is to call `idx.authenticate` passing in the verification code `{ verification: '123'}`.

> Then, handle the response from `idx.authenticate`. If the phone code was valid, the method returns a status of `Idx.Status:SUCCESS` with tokens. This status signifies that there are no more factors waiting to be enrolled and verified. If the steps described in [Set up your Okta org (for multifactor use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-multifactor-use-cases) were properly followed, the user is successfully signed in and is sent to the default sign-in home page.

Reviewer: ```suggestion
Then, handle the response from `idx.authenticate`. If the phone code was valid, the method returns a status of `Idx.Status:SUCCESS` with tokens. This status signifies that there are no more factors that are waiting to be enrolled and verified. If the steps described in [Set up your Okta org (for multifactor use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-multifactor-use-cases) were properly followed, the user is successfully signed in and is sent to the default sign-in home page.
```
